### ISSUE TITLE
Leaf node compute_layout wrong after parent node set to Display::None

### ISSUE THREAD

**Issue by dzhou121** [DioxusLabs/taffy#452]
Taffy 0.3.11

In the example below, it built a tree like this root -> parent -> child

We set root to  ```Display::None``` first, and when we set root to ```Display::Flex``` back, the size of parent is correct but not the child. I guess it's because the child isn't marked dirty so it still use the cached value?  

```
use taffy::prelude::*;

let mut taffy = taffy::Taffy::new();

let child = taffy
    .new_leaf(Style {
        size: Size {
            width: points(800.0),
            height: points(100.0),
        },
        ..Default::default()
    })
    .unwrap();

let parent = taffy
    .new_with_children(
        Style {
            size: Size {
                width: points(800.0),
                height: points(100.0),
            },
            ..Default::default()
        },
        &[child],
    )
    .unwrap();

let root = taffy
    .new_with_children(
        Style {
            ..Default::default()
        },
        &[parent],
    )
    .unwrap();
taffy.compute_layout(root, Size::MAX_CONTENT).unwrap();
assert_eq!(taffy.layout(child).unwrap().size.width, 800.0);
assert_eq!(taffy.layout(child).unwrap().size.height, 100.0);

taffy
    .set_style(
        root,
        Style {
            display: Display::None,
            ..Default::default()
        },
    )
    .unwrap();
taffy.compute_layout(root, Size::MAX_CONTENT).unwrap();
assert_eq!(taffy.layout(child).unwrap().size.width, 0.0);
assert_eq!(taffy.layout(child).unwrap().size.height, 0.0);

taffy
    .set_style(
        root,
        Style {
            ..Default::default()
        },
    )
    .unwrap();
taffy.compute_layout(root, Size::MAX_CONTENT).unwrap();
assert_eq!(taffy.layout(parent).unwrap().size.width, 800.0);
assert_eq!(taffy.layout(parent).unwrap().size.height, 100.0);
assert_eq!(taffy.layout(child).unwrap().size.width, 800.0);
assert_eq!(taffy.layout(child).unwrap().size.height, 100.0);

```

**Comment by nicoburns** [comment]
@dzhou121 Thanks for the bug report (and the test case!). Fix incoming. I'll backport it to the `0.3.x` series once it lands in `main`.

**Comment by nicoburns** [comment]
Fixed in Taffy `v0.3.12` (and `main`)

**Comment by dzhou121** [comment]
That's speedy! Thanks so much.
